### PR TITLE
Added sap_hana_rhv_* roles

### PR DESCRIPTION
**Pull request by newkit**
The sap_hana_rhv_* rules allow configuring the hypervisor and guest according to the SAP HANA on RHV Best Practices guide.

### CODE REVIEW

File: roles/sap_hana_rhv_guest/meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  # If this a Container Enabled role, provide the minimum Ansible Container version.
+  # min_ansible_container_version:
+
+  #
+  # Provide a list of supported platforms, and for each platform a list of versions.
+  # If you don't wish to enumerate all versions for a particular platform, use 'all'.
+  # To view available platforms and versions (or releases), visit:
+  # https://galaxy.ansible.com/api/v1/platforms/
+  #
+  # platforms:
```
Comment (kksat) [inline roles/sap_hana_rhv_guest/meta/main.yml:31]:
```suggestion
 platforms:
   - name: EL
     versions:
     - "8"
```
If my understanding is correct role is intended for RHEL 8 only.

File: roles/sap_hana_rhv_guest/tasks/set-tuned-profile.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+---
+- name: Create tuned profile directory /usr/lib/tuned/sap-hana
+  file:
+    path: /usr/lib/tuned/sap-hana
+    state: directory
+
+- name: Create sap-hana tuned profile
+  copy:
+    dest: "/usr/lib/tuned/sap-hana/tuned.conf"
+    content: |
```
Comment (kksat) [inline roles/sap_hana_rhv_guest/tasks/set-tuned-profile.yml:10]: Why content is used and at the same time there is a file in `./files/tuned/sap-hana/tuned.conf` exact same content? 
I think one of the above (content section or file) should remain for the ease of maintenance.

File: roles/sap_hana_rhv_guest/tasks/set-tuned-profile.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        vm.dirty_background_ratio = 10
+        vm.swappiness = 10
+
+
+- name: Create tuned profile directory /usr/lib/tuned/sap-hana-kvm-guest
+  file:
+    path: /usr/lib/tuned/sap-hana-kvm-guest
+    state: directory
+
+- name: Add haltpoll.sh for tuned sap-hana-kvm-guest
```
Comment (kksat) [inline roles/sap_hana_rhv_guest/tasks/set-tuned-profile.yml:41]: Again why use content when this file is available in the role. 
Let's keep only one.

File: roles/sap_hana_rhv_guest/tasks/set-tuned-profile.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        if [ "$1" == "start" ]; then
+            modprobe cpuidle-haltpoll force
+        fi
+
+        ## Question:  Does this also need another "if" checking to see if $1 is "stop" to unload the module?
+
+
+- name: Create sap-hana-kvm-guest tuned profile
+  copy:
+    dest: "/usr/lib/tuned/sap-hana-kvm-guest/tuned.conf"
```
Comment (kksat) [inline roles/sap_hana_rhv_guest/tasks/set-tuned-profile.yml:57]: Same here, let's keep either content or file in the role.

File: roles/sap_hana_rhv_guest/tasks/set-tuned-profile.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        kernel.sched_migration_cost_ns=500000
+        kernel.sched_min_granularity_ns=12000000
+        kernel.sched_wakeup_granularity_ns=15000000
+
+        [bootloader]
+        cmdline_saphana=skew_tick=1
+
+- name: Activate tuned profile
+  command: tuned-adm profile sap-hana-kvm-guest
+...
```
Comment (kksat) [inline roles/sap_hana_rhv_guest/tasks/set-tuned-profile.yml:86]:
```suggestion
become: true
become_user: root
```
I am not sure but root required here?

File: roles/sap_hana_rhv_hypervisor/meta/main.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  # issue_tracker_url: http://example.com/issue/tracker
+
+  # Choose a valid license ID from https://spdx.org - some suggested licenses:
+  # - BSD-3-Clause (default)
+  # - MIT
+  # - GPL-2.0-or-later
+  # - GPL-3.0-only
+  # - Apache-2.0
+  # - CC-BY-4.0
+  license: GPL-2.0-or-later
```
Comment (kksat) [inline roles/sap_hana_rhv_hypervisor/meta/main.yml:20]: Apache 2?

File: roles/sap_hana_rhv_hypervisor/meta/main.yml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  # If this a Container Enabled role, provide the minimum Ansible Container version.
+  # min_ansible_container_version:
+
+  #
+  # Provide a list of supported platforms, and for each platform a list of versions.
+  # If you don't wish to enumerate all versions for a particular platform, use 'all'.
+  # To view available platforms and versions (or releases), visit:
+  # https://galaxy.ansible.com/api/v1/platforms/
+  #
+  # platforms:
```
Comment (kksat) [inline roles/sap_hana_rhv_hypervisor/meta/main.yml:33]:
```suggestion
  platforms:
  - name: EL
    versions:
    - "8"
```
As per my understanding this role is intended for RHEL 8

File: roles/sap_hana_rhv_hypervisor/tasks/assert-configuration.yml [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+- name: "Assert - Kernel same page merging (KSM): Check if stopped"
+  assert:
+    that: "'Active: inactive (dead)' in __sap_hana_rhv_hypervisor_ksmstatus_assert.stdout"
+    fail_msg: "FAIL: ksm is running"
+    success_msg: "PASS: ksm not running"
+  ignore_errors: "{{ sap_hana_rhv_hypervisor_ignore_failed_assertion }}"
+
+- name: "Assert - Kernel same page merging (KSM) Tuning Daemon: Get status"
+  shell: systemctl status ksmtuned
```
Comment (kksat) [inline roles/sap_hana_rhv_hypervisor/tasks/assert-configuration.yml:21]: can be substituted for command, does not actually use any shell functions like piping

File: roles/sap_hana_rhv_hypervisor/tasks/assert-rhv-hooks.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+---
+- name: Check file permissions
```
Comment (kksat) [inline roles/sap_hana_rhv_hypervisor/tasks/assert-rhv-hooks.yml:2]: maybe it is better to use ansible.builtin.stat module to check file permissions?

File: roles/sap_hana_rhv_hypervisor/tasks/configuration.yml [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+  shell: systemctl stop ksmtuned
+  when: ksmtuned.rc == 0
+
+- name: Disable Kernel Samepage Merging (KSM) Tuning Daemon
+  shell: systemctl disable ksmtuned
+  when: ksmtuned.rc == 0
+
+- name: Check CPU Stepping
+  shell: lscpu | awk '/Stepping/{print $2}'
+  register: cpu_stepping_output
```
Comment (kksat) [inline roles/sap_hana_rhv_hypervisor/tasks/configuration.yml:32]: change variable name to sap_hana_rhv_hypervisor_cpu_stepping_output?

File: roles/sap_hana_rhv_hypervisor/tasks/configuration.yml [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+# tasks file for sap_hana_rhv_hypervisor
+
+- name: Test if kernel same page merging (KSM) exists
+  shell: systemctl cat ksm
+  register: ksm
+  ignore_errors: yes
+
+- name: Test if kernel same page merging (KSM) tuning daemon exists
+  shell: systemctl cat ksmtuned
+  register: ksmtuned
```
Comment (kksat) [inline roles/sap_hana_rhv_hypervisor/tasks/configuration.yml:11]: change variable name to sap_hana_rhv_hypervisor_ksmtuned?

File: roles/sap_hana_rhv_hypervisor/tasks/configuration.yml [hunk 12]
```diff
@@ -0,0 +1,131 @@
+---
+# tasks file for sap_hana_rhv_hypervisor
+
+- name: Test if kernel same page merging (KSM) exists
+  shell: systemctl cat ksm
+  register: ksm
```
Comment (kksat) [inline roles/sap_hana_rhv_hypervisor/tasks/configuration.yml:6]: change variable name to sap_hana_rhv_hypervisor_ksm

File: roles/sap_hana_rhv_hypervisor/tests/test.yml [hunk 13]
```diff
@@ -0,0 +1,6 @@
+---
+- hosts: localhost
+  remote_user: root
```
Comment (kksat) [inline roles/sap_hana_rhv_hypervisor/tests/test.yml:3]: This is a security risk, I have never seen that root will be allowed remote access. 
To mitigate this parameters become: true and become_user: root should be added to all tasks where sudo is required. 
This is my point of view.

File: roles/sap_hana_rhv_hypervisor/vars/main.yml [hunk 14]
```diff
@@ -0,0 +1,8 @@
+---
+# vars file for sap_hana_rhv_hypervisor
+#
+
+# packages to install
+sap_hana_rhv_hypervisor_packages:
```
Comment (kksat) [inline roles/sap_hana_rhv_hypervisor/vars/main.yml:6]: why this parameter is in vars, but not in defaults?

**Review by kksat — COMMENTED** [review]
I think all ansible core roles should use fully qualified role name. 
And all tasks that require sudo should have become and become_user parameters.

**Comment by sean-freeman** [comment]
As per #233 , closing this PR

**Pull request closed without merging**